test/optim/test_optim.py

```diff
@@ -1607,6 +1607,43 @@ def closure():
             opt.step(closure)
             self.assertEqual(old_param, param)
 
+    def test_step_warns_once_for_no_grads(self):
+        # not including LBFGS since it creates a zero-ed tensor in _gather_flat_grad
```

Comment:
Wait..but should it? Is this warning no longer a concern with LBFGS? My impression is that this would require special casing for LBFGS, not that the warning should not happen. I also observe Rprop is missing.